pkg/apis/node/types.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RuntimeClass defines a class of container runtime supported in the cluster.
+// The RuntimeClass is used to select or configure the container runtime that is
```

Comment:
I tried to clarify this, WDYT? The idea is that the RuntimeClass defines a "class of runtimes", but it's concrete usage in a cluster determines which runtime (that is part of that class) is run. The idea being that the runtime class could be "sandboxed", and the implementation could be gvisor in one cluster and kata containers in another. It remains to be seen how this will play out in practice...

Reply:
I like the current iteration better. It also looks like you expect some "common" Runtime Class names. May be you can mention that in the comments as well stating some common portable names are expected to be curated by the community over time?

Reply:
That is mentioned in the [KEP](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/runtime-class.md#proposed-future-enhancements), but I'd rather not speculate about changes in the API comments.